### PR TITLE
#10991: Allow mute unmute chats individually feature iOS SDK bindings.

### PR DESCRIPTION
**Pull request by lancy-mega**
Allow mute unmute chats individually feature iOS SDK bindings.

### CODE REVIEW

File: bindings/ios/MEGAPushNotificationSettings.h [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+ * This method returns a valid value only if [MEGAPushNotificationSettings isChatEnabled]
+ * returns NO and [MEGAPushNotificationSettings isChatDndEnabled] returns YES.
+ *
+ * If there's no DND mode established for the specified chat, this function returns -1.
+ * @note a DND value of 0 means the DND does not expire.
+ *
+ * @param chatId handle of the Node that identifies the chat room
+ * @return timestamp until DND mode is enabled (in seconds since the Epoch)
+ */
+- (int64_t)chatDndForChatId:(int64_t)chatId;
```
Comment (jnavarrom) [inline bindings/ios/MEGAPushNotificationSettings.h]: The method should begin with the property name of the return value, here looks like the return value is a chat.

I think we can rename this method to `timestampDndForChatId` or `chatDndTimestampForChatId`, What do you think @lancy-mega ?

File: bindings/ios/MEGARequest.h [hunk 2]
```diff
@@ -471,6 +472,15 @@ typedef NS_ENUM (NSInteger, MEGANodeAccessLevel) {
  */
 @property (readonly, nonatomic) MEGAFolderInfo *megaFolderInfo;
 
+/**
+ *  @brief Returns settings for push notifications
+ *
+ * The SDK retains the ownership of the returned value. It will be valid until the MEGARequest object is deleted.
```
Comment (jnavarrom) [inline bindings/ios/MEGARequest.h]:
```suggestion
```

This is internal for the SDK and transparent for Objective-C developers, we don't add this kind of information to the binding

File: bindings/ios/MEGARequest.h [hunk 3]
```diff
@@ -471,6 +472,15 @@ typedef NS_ENUM (NSInteger, MEGANodeAccessLevel) {
...
+ * This value is valid for these requests in onRequestFinish.
```
Comment (jnavarrom) [inline bindings/ios/MEGARequest.h]:
```suggestion
* This value is valid for these requests in onRequestFinish when the
* error code is MEGAErrorTypeApiOk:
* - [MEGASdk getPushNotificationSettingsWithDelegate] - Returns settings for push notifications
```

File: bindings/ios/Private/MEGAPushNotificationSettings+init.h [hunk 4]
```diff
@@ -0,0 +1,35 @@
+/**
+ * @file MegaPushNotificationSettings+init.h
```
Comment (jnavarrom) [inline bindings/ios/Private/MEGAPushNotificationSettings+init.h]:
```suggestion
 * @file MEGAPushNotificationSettings+init.h
```

File: bindings/ios/Private/MEGAPushNotificationSettings+init.h [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+ * @brief Private functions of MegaPushNotificationSettings
```
Comment (jnavarrom) [inline bindings/ios/Private/MEGAPushNotificationSettings+init.h]:
```suggestion
 * @brief Private functions of MEGAPushNotificationSettings
```

File: bindings/ios/MEGASdk.h [hunk 6]
```diff
@@ -7186,6 +7187,38 @@ typedef NS_ENUM(NSInteger, BusinessStatus) {
  */
 - (void)getRegisteredContacts:(NSArray<NSDictionary *> *)contacts delegate:(id<MEGARequestDelegate>)delegate;
 
+#pragma mark - Push Notification Settings
+
+/**
+ *@brief: Get push notification settings
```
Comment (jnavarrom) [inline bindings/ios/MEGASdk.h]:
```suggestion
 *@brief Get push notification settings
```

File: bindings/ios/MEGASdk.h [hunk 7]
```diff
@@ -7186,6 +7187,38 @@ typedef NS_ENUM(NSInteger, BusinessStatus) {
...
+ * Valid data in the MEGARequest object received in onRequestFinish when the error code
+ * is MEGAErrorTypeApiOk:
+ * - [MEGARequest megaPushNotificationSettings] Returns settings for push notifications
+ *
+ *@param delegate MEGARequestDelegate to track this request
+ */
+- (void)getPushNotificationSettingsWithDelegate:(id<MEGARequestDelegate>)delegate;
+
+/**
+ *@brief: Set push notification settings.
```
Comment (jnavarrom) [inline bindings/ios/MEGASdk.h]:
```suggestion
 *@brief Set push notification settings.
```

File: bindings/ios/MEGASdk.h [hunk 8]
```diff
@@ -7186,6 +7187,38 @@ typedef NS_ENUM(NSInteger, BusinessStatus) {
...
+ * The associated request type with this request is MEGARequestTypeSetAttrUser
+ * - [MEGARequest paramType] - Returns the attribute type MEGAUserAttributePushSettings
+ *
+ * Valid data in the MEGARequest object received in onRequestFinish when the error code
+ * is MEGAErrorTypeApiOk:
+ * - [MEGARequest megaPushNotificationSettings] Returns settings for push notifications
+ *
+ *@param delegate MEGARequestDelegate to track this request
+ */
+- (void)getPushNotificationSettingsWithDelegate:(id<MEGARequestDelegate>)delegate;
```
Comment (jnavarrom) [inline bindings/ios/MEGASdk.h]: We need to add the method to get the push notifications without the needed of a delegate. 

`- (void)getPushNotification`

File: bindings/ios/MEGASdk.h [hunk 9]
```diff
@@ -7186,6 +7187,38 @@ typedef NS_ENUM(NSInteger, BusinessStatus) {
...
+ * - [MEGARequest paramType] - Returns the attribute type MEGAUserAttributePushSettings
+ *
+ * Valid data in the MEGARequest object received in onRequestFinish when the error code
+ * is MEGAErrorTypeApiOk:
+ * - [MEGARequest megaPushNotificationSettings] Returns settings for push notifications
+ *
+ *@param pushNotificationSettings Push notification settings of the user. (An instance of MEGAPushNotificationSettings).
+ *@param delegate MEGARequestDelegate to track this request
+ */
+- (void)setPushNotificationSettings:(MEGAPushNotificationSettings *)pushNotificationSettings
```
Comment (jnavarrom) [inline bindings/ios/MEGASdk.h]: Same as before

`- (void)setPushNotificationSettings`

**Comment by lancy-mega** [comment]
@jnavarrom : I have handled all the PR comments. Please have a look. Thanks.

**Pull request merged**